manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/services/BridgeServiceImpl.java

```diff
@@ -232,53 +238,62 @@ public Bridge updateBridgeStatus(ResourceStatusDTO statusDTO) {
             return bridge;
         }
         Operation operation = bridge.getOperation();
+        if (Objects.isNull(operation)) {
+            LOGGER.info("Update for Bridge with id '{}' was discarded. The Bridge Operation is null.", bridge.getId());
+            return bridge;
+        }
+
         List<Condition> conditions = bridge.getConditions();
+        // Set the updated conditions to the existing Manager conditions to begin; then copy in the new Operator conditions
         List<Condition> updatedConditions = conditions.stream().filter(c -> c.getComponent() == ComponentType.MANAGER).collect(Collectors.toList());
         statusDTO.getConditions().forEach(c -> updatedConditions.add(Condition.from(c, ComponentType.SHARD)));
+
+        // Don't do anything if Conditions are unchanged from our previous state.
+        ManagedResourceStatusV2 status = StatusUtilities.getManagedResourceStatus(operation, conditions);
+        ManagedResourceStatusV2 originalStatus = StatusUtilities.getManagedResourceStatus(operation, updatedConditions);
+        if (Objects.equals(status, originalStatus)) {
+            LOGGER.info("Update for Bridge with id '{}' was discarded. The ManagedResourceStatus reflected by the Conditions is unchanged.", bridge.getId());
+            return bridge;
+        }
+
         bridge.setConditions(updatedConditions);
 
         switch (operation.getType()) {
             case CREATE:
-                if (isOperationComplete(updatedConditions)) {
+                if (ConditionUtilities.isOperationComplete(updatedConditions)) {
                     if (Objects.isNull(bridge.getPublishedAt())) {
                         bridge.setPublishedAt(ZonedDateTime.now(ZoneOffset.UTC));
-                        // TODO: record metrics with MetricsService
-                        // metricsService.onOperationComplete(bridge, MetricsOperation.MANAGER_RESOURCE_PROVISION);
+                        metricsService.onOperationComplete(bridge, MetricsOperation.MANAGER_RESOURCE_PROVISION);
                     }
+                } else if (ConditionUtilities.isOperationFailed(updatedConditions)) {
+                    metricsService.onOperationFailed(bridge, MetricsOperation.MANAGER_RESOURCE_PROVISION);
                 }
                 break;
 
             case UPDATE:
-                // TODO: record metrics with MetricsService
-                // metricsService.onOperationComplete(bridge, MetricsOperation.MANAGER_RESOURCE_MODIFY);
+                if (ConditionUtilities.isOperationComplete(updatedConditions)) {
+                    metricsService.onOperationComplete(bridge, MetricsOperation.MANAGER_RESOURCE_MODIFY);
```

Comment:
@manstis For consistency, should we rename this to `MANAGER_RESOURCE_UPDATE`. I'm aware that this will impact `v1` as well, but if there is nothing relying on it, I don't think it's a problem

Reply:
Changed in a new commit. I also changed the _tag value_ to `update` too (as opposed to just changing the name of the enumeration entry). This means we have two _operations_ with the same _tag value_ but since they're used on different metrics I don't think it will cause a problem.